shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -171,6 +175,15 @@ - (instancetype)initWithViewController:(FlutterViewController*)viewController {
   return self;
 }
 
+- (BOOL)isFirstResponder {
+  FlutterAppDelegate* appDelegate = (FlutterAppDelegate*)[NSApp delegate];
+  return [appDelegate.mainFlutterWindow firstResponder] == self;
```

Comment:
This seems like it won't really work if someone is embedding a FlutterViewController along side some other application. I'm not sure if that's a case we care about supporting on macOS right now, but wouldn't it be safer/more robust to trace the responder chain from the VC we were passed in the initializer?